lib/Gateway.js

```diff
@@ -1366,14 +1366,8 @@ Gateway.prototype.discoverValue = function (node, vId) {
 
         Object.assign(cfg.discovery_payload, sensor.props || {})
 
-        // Discovered valueId.unit also exists. therefore added both unit and units to be
-        // checked, but only one used
-        if (valueId.unit) {
-          // https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/scales.json
-          cfg.discovery_payload.unit_of_measurement = valueId.unit
-        } else if (valueId.units) {
-          cfg.discovery_payload.unit_of_measurement = valueId.unit
-        }
+        // https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/scales.json
+        cfg.discovery_payload.unit_of_measurement = valueId.unit
```

Comment:
@billiaz I would keep the if here, or this could be undefined and hass will throw an error